src/components/Mentees/Mentees.js

```diff
@@ -0,0 +1,524 @@
+        </div>
+      )}
+    </div>
+  );
+};
+
+class Mentees extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      mentees: [],
+      uid: "",
+      open: false,
+      menteeKey: "",
+      order: "desc",
+      orderBy: "name",
+      selected: [],
+      page: 0,
+      rowsPerPage: 25,
+      menteeData: {},
+      filterApplied: false,
+      specialty: ""
+    };
+  }
+  handleRequestSort = (event, property) => {
+    const orderBy = property;
+    let order = "desc";
+
+    if (this.state.orderBy === property && this.state.order === "desc") {
+      order = "asc";
+    }
+
+    this.setState({ order, orderBy });
+  };
+
+  handleClick = (event, key, menteeData) => {
+    this.state.menteeKey === key
+      ? this.setState({ menteeKey: "", menteeData: {} })
+      : this.setState({ menteeKey: key, menteeData: menteeData });
+  };
+
+  handleChangePage = (event, page) => {
+    this.setState({ page });
+  };
+
+  /**
+   * handleChange – the handleChange sets the specialty wrote in the state
+   * @param {Object} the object name and event
+   * @return {void}
+   */
+  handleChange = event => {
+    const { target } = event;
+    const { value, name } = target;
+    this.setState({
+      [name]: value
+    });
+  };
+
+  handleChangeRowsPerPage = event => {
+    this.setState({ rowsPerPage: event.target.value });
+  };
+
+  isSelected = key => this.state.menteeKey === key;
+
+  arrayToObject = array =>
+    array.reduce((obj, item) => {
+      obj[item[0]] = item[1];
+      return obj;
+    }, {});
+
+  filterByLocation = () => {
+    const location = this.state.location;
+    if (location.trim().length > 0) {
+      const menteesByLocation = this.arrayToObject(
+        Object.entries(this.state.menteesMirror).filter(mentee =>
+          mentee[1].location.toLowerCase().includes(location.toLowerCase())
+        )
+      );
+      this.setState({
+        mentees: menteesByLocation,
+        filterApplied: true
+      });
+    }
+  };
+
+  getAllMentees = () => {
+    getMentees()
+      .then(snapshot => {
+        this.setState({
+          mentees: snapshot.val(),
+          menteesMirror: snapshot.val(),
+          filterApplied: false
+        });
+      })
+      .catch(
+        this.setState({
+          mentees: "",
+          menteesMirror: "",
+          filterApplied: false
+        })
+      );
+  };
+
+  componentDidMount() {
+    if (this.props.authUser) {
+      this.unregisterObserver = this.getAllMentees();
+    }
+  }
+  componentDidUpdate(prevProps) {
+    if (this.props !== prevProps) {
+      this.getAllMentees();
+    }
+  }
+  componentWillUnmount() {
+    this.unregisterObserver = null;
+  }
+
+  render() {
+    const { classes, authUser } = this.props;
+    const { from } = this.props.location.state || {
+      from: { pathname: "/nofound" }
+    };
+    const name = this.props.authUser ? this.props.authUser.userName : "";
```

Comment:
If we're not using `name` we can delete it.

Reply:
Good eye! Done.